src/base/rss/rss_session.h

```diff
@@ -0,0 +1,153 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2017  Vladimir Golovnev <glassez@yandex.ru>
+ * Copyright (C) 2010  Christophe Dumez <chris@qbittorrent.org>
+ * Copyright (C) 2010  Arnaud Demaiziere <arnaud@qbittorrent.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+/*
+ * RSS Session configuration file format (JSON):
+ *
+ * =============== BEGIN ===============
+ * {
+ *     "folder1": {
+ *         "subfolder1": {
+ *             "Feed name (Alias)": "http://some-feed-url1",
+ *             "http://some-feed-url2": ""
+ *         },
+ *         "subfolder2": {},
+ *         "http://some-feed-url3": "",
+ *         "Feed name (Alias)": {
+ *             "url": "http://some-feed-url4",
+ *         }
+ *     },
+ *     "folder2": {},
+ *     "folder3": {}
+ * }
+ * ================ END ================
+ *
+ * 1.   Document is JSON object (the same as Folder)
+ * 2.   Folder is JSON object (keys are Item names, values are Items)
+ * 3.1. Feed is JSON object (keys are property names, values are property values; 'url' is required)
+ * 3.2. (Reduced format) Feed is JSON string (string is URL unless it's empty, otherwise we take Feed URL from name)
+ */
+
+#include <QHash>
+#include <QObject>
+#include <QPointer>
+#include <QStringList>
+#include <QTimer>
+
+class QThread;
+class Application;
+class AsyncFileStorage;
+
+namespace RSS
+{
+    class Item;
+    class Feed;
+    class Folder;
+
+    class Session: public QObject
+    {
+        Q_OBJECT
+
+        friend class ::Application;
```

Comment:
Isn't it a good idea to keep things the similar? All of our other singletons use the initInstance()/freeInstance()/instance() triplet. Unless you want to solve a technical problem...